app/controllers/schools_controller.rb

```diff
@@ -0,0 +1,39 @@
+class SchoolsController < ApplicationController
+  def show
+    @school = School.find(params[:id])
+  end
+
+  def new; end
+
+  def create
+    validate_opened_date
+    return if performed?
+
+    @school = School.new(params.require(:school).permit(:name, :opened, :school_type))
```

Comment:
Better to move params.require(:school).permit(:name, :opened, :school_type) to a private method. It would change a lot as more school attributes are added. Also, that way it could be reused for another action.